File: iot-rust/src/core/runtime/dependencies.rs
use crate::core::device::gateways::config_storage::ConfigStorage;
use crate::core::device::gateways::device_id_generator::DeviceIdGenerator;

pub struct Dependencies<'a, C: ConfigStorage, D: DeviceIdGenerator> {
    pub config_store: &'a C,
    pub device_id_generator: &'a D,
}

impl<'a, C: ConfigStorage, D: DeviceIdGenerator> Dependencies<'a, C, D> {
    pub fn new(config_store: &'a C, device_id_generator: &'a D) -> Self {
        Self { config_store, device_id_generator }
    }
}

File: packages/iot/core/dependencies.go
package core

import (
	ble_gateways "github.com/carpe-platform/iot-golang/core/ble/gateways"
	device_gateways "github.com/carpe-platform/iot-golang/core/device/gateways"
	device_generators "github.com/carpe-platform/iot-golang/core/device/generators"
)

type Dependencies struct {
	ConfigGateway       device_gateways.ConfigGateway
	DeviceIdGenerator   device_generators.DeviceIdGenerator
	BluetoothAdvertising ble_gateways.BluetoothAdvertising
}
